mlir/test/IR/recursive-distinct-attr.mlir

```diff
@@ -0,0 +1,9 @@
+// RUN: mlir-opt -emit-bytecode %s | mlir-opt --mlir-print-debuginfo | FileCheck %s
+
+// CHECK: distinct[0]
+// CHECK-NOT: distinct[1]
+#attr_ugly = #test<attr_ugly begin distinct[0]<> end>
+#attr_ugly1 = #test<attr_ugly begin #attr_ugly end>
+
+module attributes {test.alias = #attr_ugly} {
+} loc(fused<#attr_ugly1>[])
```

Comment:
Back to a question I had before (maybe I missed the answer): why is the bug only triggered when the alias is used in a location?